polkadot/xcm/pallet-derivatives/src/lib.rs

```diff
@@ -0,0 +1,403 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The purpose of the `pallet-derivatives` is to cover the following derivative asset support
+//! scenarios:
+//! 1. The `pallet-derivatives` can serve as an API for creating and destroying derivatives.
+//! 2. It can store a mapping between the foreign original ID (e.g., XCM `AssetId` or `(AssetId,
+//!    AssetInstance)`) and the local derivative ID.
+//!
+//! The scenarios can be combined.
+//!
+//! ### Motivation
+//!
+//! The motivation differs depending on the scenario in question.
+//!
+//! #### The first scenario
+//!
+//! The `pallet-derivatives` can be helpful when another pallet, which hosts the derivative assets,
+//! doesn't provide a good enough way to create new assets in the context of them being derivatives.
+//!
+//! This mainly concerns derivative NFT collections because they should be creatable by an
+//! unprivileged user, in contrast to how fungible derivative assets are usually registered using a
+//! privileged origin (root or some collective).
+//!
+//! Fungible derivatives require a privileged origin to be registered since they could be used as
+//! fee payment assets. Conversely, Derivative NFT collections contain unique derivative objects
```

Comment:
This is, not true? While they could be used for fee payment, they can only be used like that if they're present in a liquidity pool with DOT in the case of AH. Other chains could configure it differently and might even need them to be registered from a privileged origin, but it's not at all mandatory.